util/deadlockhistory/deadlock_history_test.go

```diff
@@ -277,3 +277,27 @@ func (s *testDeadlockHistorySuite) TestErrDeadlockToDeadlockRecord(c *C) {
 	expectedRecord.OccurTime = record.OccurTime
 	c.Assert(record, DeepEquals, expectedRecord)
 }
+
+func dummyRecord() *DeadlockRecord {
+	return &DeadlockRecord{}
+}
+
+func (s *testDeadlockHistorySuite) TestResize(c *C) {
+	h := NewDeadlockHistory(2)
+	h.Push(dummyRecord()) // id=1 inserted
+	h.Push(dummyRecord()) // id=2 inserted,
+	h.Push(dummyRecord()) // id=3 inserted, id=1 is removed
+	c.Assert(len(h.GetAll()), Equals, 2)
+	c.Assert(h.GetAll()[0].ID, Equals, uint64(2))
+	c.Assert(h.GetAll()[1].ID, Equals, uint64(3))
+	h.Resize(3)
+	h.Push(dummyRecord()) // id=4 inserted
+	c.Assert(len(h.GetAll()), Equals, 3)
+	c.Assert(h.GetAll()[0].ID, Equals, uint64(2))
+	c.Assert(h.GetAll()[1].ID, Equals, uint64(3))
+	c.Assert(h.GetAll()[2].ID, Equals, uint64(4))
+	h.Resize(2) // id=2 removed
+	c.Assert(len(h.GetAll()), Equals, 2)
+	c.Assert(h.GetAll()[0].ID, Equals, uint64(3))
+	c.Assert(h.GetAll()[1].ID, Equals, uint64(4))
```

Comment:
Consider also cover resizing to 0, resizing from 0 to non-zero, and assert the value of `head` and `size`.